core/trino-main/src/main/java/io/trino/eventlistener/EventListenerManager.java

```diff
@@ -133,6 +141,13 @@ private static Map<String, String> loadEventListenerProperties(File configFile)
     }
 
     public void queryCompleted(Function<Boolean, QueryCompletedEvent> queryCompletedEventProvider)
+    {
+        try (TimeStat.BlockTimer ignored = queryCompletedTime.time()) {
+            doQueryCompleted(queryCompletedEventProvider);
+        }
+    }
+
+    private void doQueryCompleted(Function<Boolean, QueryCompletedEvent> queryCompletedEventProvider)
```

Comment:
sure. I was just checking how it would render.
I'm not sure about the long-term. It makes the methods interleaved in the sense that it's:
- public
- private
- public
- private
- ...

According to uncle Bob's clean code, that's actually bad code. Not sure what the guidelines are here.

Reply:
You could batch methods:
 - public methods
 - private "do" methods
 - public `@Managed` methods - this is actual a technical thing that is "public". They are not a real public API of this class.
 
 
 We are not strictly following clean code.